### DISCUSSION TITLE
Problem with hack 9FUSY

### DISCUSSION THREAD

**Discussion by piotrwozniak** [roleoroleo/yi-hack-Allwinner#230] [Q&A]
Precisely speaking: 9FUSY314OIJM

I read that I should use allwinner and indeed only this soft do anything else than nothing. After restart camera with allwinner I notice blue light blinking about 5sec, then yellow stable light about 1min, than yellow blink and „waiting to connect”. As I suppose last blink is ordinary restart of the camera. My question is do you know how successfully hack my camera? Is it too new to hack it? Does your soft is compatible with this camera somehow? Geneally speaking help!

**Comment by rrampone** [comment]
I have the same issue exactly with a 9FUSY32D1S, Yi Home 2 AI+  apparently doesn't recognise the files as a new firmware

**Reply by JohnCanty** [reply]
Exact same symptoms here, I attempted everything I could find, downgrade wouldn't go either. 9FUSY31ER4

**Comment by roleoroleo** [comment]
The serial number is not enough to know the exact model.
Check fw version with the app.

**Reply by JohnCanty** [reply]
I didn't install the app. I ended up returning the cameras. I was interested in the offline functionality only. I think the cameras that are shipping now have  the AI+ badge on them as well. So they could be yet another variant.

**Reply by piotrwozniak** [reply]
@roleoroleo could you generally describe the problem with hacking this type of cameras? Can it be handled somehow or it is some kind of hardware issue and I should do as @JohnCanty did? I do not know what to do but I want to make a survilliance system for my sister till the end of march :) i like this cameras and I cannot be 100% sure if I buy next 2 cams they work.

**Comment by roleoroleo** [comment]
Yi sells these cameras with the same name and different hardware.
Every year a new model comes out: externally it's exactly identical to the previous ones but contains a completely different hardware: cpu, sensor, wifi, etc ...
You cannot know the model before you buy it.
And the serial number is not enough. You must to know the fw version also.

**Reply by piotrwozniak** [reply]
How to check fw? And if I know it, could it be possible to adjust a software also for my cameras? I can open cameras and check everything I can but I have no idea is it reasonable?

**Reply by roleoroleo** [reply]
You can check the firmware version with the Yi app.
If your cam is not supported by any firmware (v4, mstar, allwinner, allwinner-v2), maybe it's possible to create a new fw.
But it depends on the platform.

**Reply by piotrwozniak** [reply]
My actual Firmware Version is 8.2.0.0A_201912270941 and I can update to 8.2.0.0A_202012041528 but I have not beed updated yet. Having this info is it possible to say more about my cameras? Should one of the custom firmware be working?

**Comment by roleoroleo** [comment]
Your cam is a y20ga and it's supported by this firmware.
Try again your hack procedure.

**Reply by piotrwozniak** [reply]
I did it! It worked. Thank you! I used sdwifi.tar.gz to change wifi credentials but blue lamp still blinking. How long does should take? Did you noticed any problems with this scripts?

**Reply by roleoroleo** [reply]
You should use this script only when the cam repeats the connection message every boot.
If it isn't your case, the standard procedure is:
-  configure wifi with the original app
-  apply hack

sdwifi.tar.gz is fast, 1 minute...
There is only 1 problem with this script: you can't use it if the cam is not hacked.

**Reply by piotrwozniak** [reply]
What I did in details: 
I hacked the camera but I did not want to input during standard configuration Wifi credentials to my private wifi because I dont want to pass it to China. Next I created temp hotspot connection on my phone and setup cam using this temp wifi credentials. When blue establish I noticed that yi hack http page is available on 8080. Next step was to a) disable chineese servers and b) change wifi credentials but in the end I did only b) - i used tar.gz to change credentials. After reboot I have blinking blue so I dont know why I cannot connect to my target wifi. I noticed what you wrote below, that I cannot use tar.gz when cam is not hacked and that it is connected with fact that different sh script is run when cam is hacked. Is it some gap in my way of connection camera? What can I do with blinking blue right now?

**Reply by roleoroleo** [reply]
Probably you should use a different wifi script, similar to this one:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/65

**Reply by piotrwozniak** [reply]
Did you notice any problems with rstp or other type of streamming video? I tested this cameras on motioneye and synology survilliance and in both tools noticed lack of video and gray screens. Is it still in beta version or maybe my hardware fails?

**Reply by roleoroleo** [reply]
It depends on the type of problem.
RTSP server is not perfect but it works quite well.
I will release ASAP a new version with some improvements.

**Comment by piotrwozniak** [comment]
I am looking forward! 🤓Thank you.

**Comment by piotrwozniak** [comment]
For your consideration 1) it would be good to have some kind of calculator which gives an answer what kind of soft should be used to hack and 2) changing wifi id/pass directly from flex would be very useful - update possible after input password.
